superbuild/CMakeLists.txt

```diff
@@ -868,25 +912,70 @@ endif()
 
 # Stage1 job to build bootstrap compiler for building required Clang/LLVM libraries and ISPC itself.
 if (PGO OR NOT SKIP_STAGE2_DEPS)
-    ExternalProject_Add(stage1
-        DOWNLOAD_COMMAND ""
-        UPDATE_COMMAND ""
-        SOURCE_DIR llvm-source
-        SOURCE_SUBDIR llvm
-        PREFIX bstage1
-        STAMP_DIR bstage1/s
-        BINARY_DIR bstage1/b
-        TMP_DIR bstage1/t
-        INSTALL_DIR stage2
-        CMAKE_GENERATOR Ninja
-        CMAKE_ARGS ${LLVM_STAGE1_CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/stage2
-        BUILD_IN_SOURCE OFF
-        BUILD_COMMAND ""
-        BUILD_COMMAND ${NINJA_COMMAND_WITH_ENV_STAGE1}
-        INSTALL_COMMAND ${NINJA_COMMAND_WITH_ENV_STAGE1} install
-        )
-    # Note: We may rebuild instead stage2 clang/lld with stage1 toolchain to
-    # generate more performant stage2 compiler.
+
+    # It may be empty when PREBUILT_STAGE1_PATH provided otherwise we need to depend on the stage1 external project
+    list(APPEND LLVM_STAGE2_DEPS)
+
+    if (NOT SKIP_STAGE1_DEPS)
+        ExternalProject_Add(stage1
+            DOWNLOAD_COMMAND ""
+            UPDATE_COMMAND ""
+            SOURCE_DIR llvm-source
+            SOURCE_SUBDIR llvm
+            PREFIX bstage1
+            STAMP_DIR bstage1/s
+            BINARY_DIR bstage1/b
+            TMP_DIR bstage1/t
+            INSTALL_DIR stage2
+            CMAKE_GENERATOR Ninja
+            CMAKE_ARGS ${LLVM_STAGE1_CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/stage2
+            BUILD_IN_SOURCE OFF
+            BUILD_COMMAND ""
+            BUILD_COMMAND ${NINJA_COMMAND_WITH_ENV_STAGE1}
+            INSTALL_COMMAND ${NINJA_COMMAND_WITH_ENV_STAGE1} install
+            )
+        # Note: We may rebuild instead stage2 clang/lld with stage1 toolchain to
+        # generate more performant stage2 compiler.
+
+        list(APPEND LLVM_STAGE2_DEPS stage1)
+
+        if (BUILD_STAGE1_TOOLCHAIN_ONLY)
+            # Custom target to install stage1 toolchain
+            # It may be slightly confusing because stage1 toolchain placed under stage2 folder
+            if (NOT STAGE1_TOOLCHAIN_INSTALL_PREFIX STREQUAL "NO")
+                add_custom_target(install-stage1-toolchain
```

Comment:
Should this target depend on `stage1` as `package-stage1` target?
I would also use `WORKING_DIRECTORY` here.

Reply:
Yes, it should. Let me reconsider that, I probably need to clean up all `STAGE*_PREFIX` in favor of using `CMAKE_INSTALL_PREFIX` that, I think, let me remove this extra custom target.